### DISCUSSION TITLE
Support CompletableFuture as a response type of a RequestFilter method

### DISCUSSION THREAD

**Discussion by sdelamo** [micronaut-projects/micronaut-core#11275] [Ideas]
I think it would be good to support `CompletableFuture<HttpResponse>` as a [return type of `@RequestFilter` method.](https://github.com/micronaut-projects/micronaut-core/blob/4.7.x/http/src/main/java/io/micronaut/http/annotation/RequestFilter.java#L63-L72).

**Comment by sdelamo** [comment]
thoughts @yawkat @dstepanov ?

**Reply by yawkat** [reply]
sure why not

**Reply by dstepanov** [reply]
I think it's supported

**Reply by yawkat** [reply]
Another undocumented feature? haha

**Reply by sdelamo** [reply]
it does not work. I tried in https://github.com/micronaut-projects/micronaut-security/pull/1826 and it does not work.

**Reply by sdelamo** [reply]
well, it works with `CompletableFuture<@Nullable HttpResponse<?>> `. It does not work with `CompletableFuture< HttpResponse<?>>`. That it is not documented.

**Comment by sdelamo** [comment]
I created a PR to document it https://github.com/micronaut-projects/micronaut-core/pull/11285
